remoting/zookeeper/listener.go

```diff
@@ -81,6 +76,56 @@ func (l *ZkEventListener) ListenServiceNodeEvent(zkPath string, listener remotin
 	}(zkPath, listener)
 }
 
+// ListenConfigurationEvent listen a path node event
+func (l *ZkEventListener) ListenConfigurationEvent(zkPath string, listener remoting.DataListener) {
+	// listen l service node
+	l.wg.Add(1)
+	go func(zkPath string, listener remoting.DataListener) {
+		var eventChan = make(chan zk.Event, 16)
+		l.Client.RegisterEvent(zkPath, eventChan)
+		for {
+			select {
+			case event := <-eventChan:
+				logger.Infof("[ZkEventListener]Receive configuration change event:%#v", event)
+				if event.Type == zk.EventNodeChildrenChanged || event.Type == zk.EventNotWatching {
+					continue
+				}
+				// 1. Re-set watcher for the zk node
+				_, _, _, err := l.Client.Conn.ExistsW(event.Path)
+				if err != nil {
+					logger.Warnf("[ZkEventListener]Re-set watcher error, the reason is %+v", err)
+				}
+
+				action := remoting.EventTypeAdd
+				var content string
+				if event.Type == zk.EventNodeDeleted {
+					action = remoting.EventTypeDel
+				} else {
+					// 2. Try to get new configuration value of the zk node
+					// Notice: The order of step 1 and step 2 cannot be swapped, if you get value(with timestamp t1)
+					// before re-set the watcher(with timestamp t2), and some one change the data of the zk node after
+					// t2 but before t1, you may get the old value, and the new value will not trigger the event.
+					contentBytes, _, err := l.Client.Conn.Get(event.Path)
+					if err != nil {
```

Comment:
If an error occurs, need continue execution?

Reply:
you're right, it's better do not continue the execution, I'll fix it.